globals.js

```diff
@@ -0,0 +1,939 @@
+	SVGTransform: { canOverride: false },
+	SVGTransformList: { canOverride: false },
+	SVGTSpanElement: { canOverride: false },
+	SVGUnitTypes: { canOverride: false },
+	SVGUseElement: { canOverride: false },
+	SVGViewElement: { canOverride: false },
+	TaskAttributionTiming: { canOverride: false },
+	Text: { canOverride: false },
+	TextDecoder: { canOverride: false },
+	TextEncoder: { canOverride: false },
+	TextEvent: { canOverride: false },
+	TextMetrics: { canOverride: false },
+	TextTrack: { canOverride: false },
+	TextTrackCue: { canOverride: false },
+	TextTrackCueList: { canOverride: false },
+	TextTrackList: { canOverride: false },
+	TimeRanges: { canOverride: false },
+	toolbar: { canOverride: false },
+	top: { canOverride: false },
+	Touch: { canOverride: false },
+	TouchEvent: { canOverride: false },
+	TouchList: { canOverride: false },
+	TrackEvent: { canOverride: false },
+	TransitionEvent: { canOverride: false },
+	TreeWalker: { canOverride: false },
+	UIEvent: { canOverride: false },
+	URL: { canOverride: false },
+	URLSearchParams: { canOverride: false },
+	ValidityState: { canOverride: false },
+	visualViewport: { canOverride: false },
+	VisualViewport: { canOverride: false },
+	VTTCue: { canOverride: false },
+	WaveShaperNode: { canOverride: false },
+	WebAssembly: { canOverride: false },
+	WebGL2RenderingContext: { canOverride: false },
+	WebGLActiveInfo: { canOverride: false },
+	WebGLBuffer: { canOverride: false },
+	WebGLContextEvent: { canOverride: false },
+	WebGLFramebuffer: { canOverride: false },
+	WebGLProgram: { canOverride: false },
+	WebGLQuery: { canOverride: false },
+	WebGLRenderbuffer: { canOverride: false },
+	WebGLRenderingContext: { canOverride: false },
+	WebGLSampler: { canOverride: false },
+	WebGLShader: { canOverride: false },
+	WebGLShaderPrecisionFormat: { canOverride: false },
+	WebGLSync: { canOverride: false },
+	WebGLTexture: { canOverride: false },
+	WebGLTransformFeedback: { canOverride: false },
+	WebGLUniformLocation: { canOverride: false },
+	WebGLVertexArrayObject: { canOverride: false },
+	WebSocket: { canOverride: false },
+	WheelEvent: { canOverride: false },
+	window: { canOverride: false },
+	Window: { canOverride: false },
+	Worker: { canOverride: false },
+	WritableStream: { canOverride: false },
+	XMLDocument: { canOverride: false },
+	XMLHttpRequest: { canOverride: false },
+	XMLHttpRequestEventTarget: { canOverride: false },
+	XMLHttpRequestUpload: { canOverride: false },
+	XMLSerializer: { canOverride: false },
+	XPathEvaluator: { canOverride: false },
+	XPathExpression: { canOverride: false },
+	XPathResult: { canOverride: false },
+	XSLTProcessor: { canOverride: false },
+};
+
+const commonjsGlobals = {
+	exports: { canOverride: true },
+	global: { canOverride: false },
+	module: { canOverride: false },
+	require: { canOverride: false },
+};
+
+const esXXXXGlobals = {
+	Array:              	{ canOverride: false, in: [BUILTIN, ES5, ES2015, ES2017] },
```

Comment:
I would rather have separate objects for each language version with globals that are unique to that version and then dynamically mix them together.

`const es2017globals = ...`

Reply:
Thanks for reviewing!

How would you handle knowing when a global has ceased to exist under that approach? (Or are we assuming that once in, a global will always exist, even if deprecated?)

Reply:
> Or are we assuming that once in, a global will always exist, even if deprecated?

Yup